### PR TITLE
Update to Java 17

### PR DESCRIPTION
**Pull request by mrie04**
https://github.com/metasfresh/metasfresh/issues/14856

### CODE REVIEW

**Review by metas-ts — APPROVED** [review]

File: backend/de.metas.business/src/test/java/de/metas/project/service/ProjectServiceTest.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
 import org.compiere.model.I_C_ProjectType;
 import org.compiere.model.I_C_UOM;
 import org.compiere.model.I_R_StatusCategory;
-import org.elasticsearch.common.collect.List;
+import org.elasticsearch.core.List;
```
Comment (teosarca) [inline backend/de.metas.business/src/test/java/de/metas/project/service/ProjectServiceTest.java:47]: i think that's not used  at all

File: backend/de.metas.business/src/test/java/de/metas/project/workorder/conflicts/ResourceAllocations_TestCase_ActualAndSimulation_Test.java [hunk 2]
```diff
@@ -1,50 +1,37 @@
...
-import static io.github.jsonSnapshot.SnapshotMatcher.validateSnapshots;
 import static org.assertj.core.api.Assertions.*;
 
+@ExtendWith(SnapshotExtension.class)
 public class ResourceAllocations_TestCase_ActualAndSimulation_Test
 {
-	@BeforeAll
-	static void beforeAll()
-	{
-		start(AdempiereTestHelper.SNAPSHOT_CONFIG, SnapshotFunctionFactory.newFunction());
```
Comment (teosarca) [inline backend/de.metas.business/src/test/java/de/metas/project/workorder/conflicts/ResourceAllocations_TestCase_ActualAndSimulation_Test.java:30]: @mrie04 @metas-ts is there any place where we config that config and SnapshotFunctionFactory.newFunction() to be used?

File: misc/services/procurement-webui/procurement-webui-backend/src/main/java/de/metas/procurement/webui/Application.java [hunk 3]
```diff
@@ -81,12 +75,12 @@ public TaskExecutor asyncCallsTaskExecutor()
...
-	@Bean
-	public Docket docket()
-	{
-		return new Docket(DocumentationType.OAS_30)
-				.select()
-				.paths(PathSelectors.any())
-				.build();
-	}
+	// @Bean
+	// public Docket docket()
```
Comment (teosarca) [inline misc/services/procurement-webui/procurement-webui-backend/src/main/java/de/metas/procurement/webui/Application.java:79]: @mrie04 @metas-ts saw that swagger for procurement is commented out. is this a conscious decision or just a mistake?

Comment (mrie04) [reply misc/services/procurement-webui/procurement-webui-backend/src/main/java/de/metas/procurement/webui/Application.java:79]: this one was a mistake, i'll adjust

File: backend/de.metas.business/src/test/java/de/metas/project/service/ProjectServiceTest.java (hunk 1 above)
Comment (mrie04) [reply backend/de.metas.business/src/test/java/de/metas/project/service/ProjectServiceTest.java:47]: it's used in line 76

File: backend/de.metas.business/src/test/java/de/metas/project/workorder/conflicts/ResourceAllocations_TestCase_ActualAndSimulation_Test.java (hunk 2 above)
Comment (mrie04) [reply backend/de.metas.business/src/test/java/de/metas/project/workorder/conflicts/ResourceAllocations_TestCase_ActualAndSimulation_Test.java:30]: in the new snapshot implementation i use snapshot.properties config files for configuration ( serializer.orderedJson=au.com.origin.snapshots.jackson.serializers.DeterministicJacksonSnapshotSerializer ). It doesn't use SnapshotFunctionFactory atm

**Pull request merged**